### ISSUE TITLE
[gatsby-remark-copy-linked-files] Add config option for destination directory

### ISSUE THREAD

**Issue by Undistraction** [gatsbyjs/gatsby#2779]
Would you accept a PR for adding a config option to gatsby-remark-copy-linked-files that would allow the destination directory to be selected?

The config would look like this:

```
plugins: [
  {
    resolve: `gatsby-transformer-remark`,
    options: {
      plugins: [
        { 
          resolve: 'gatsby-remark-copy-linked-files',
          options: {
            ignoreFileExtensions: [],
            destinationDir: 'downloads'
          },
        }
      ]
    }
  }
]
```

This would allow a user to have their downloads served from the specified directory instead of the root:

```
example.com/downloads/alpha.pdf
example.com/downloads/bravo.pdf
example.com/downloads/charlie.pdf
```

Effectively whatever dir or path is specificed in `destinationDir` will be placed inside the public dir. It looks like it should be pretty easy to add - just a matter of swapping out the hardcoded 'public' dir for a config option with a default to `public` and some validation of the config option.

**Comment by KyleAMathews** [comment]
Yeah, this seems reasonable.

**Comment by giraffesyo** [comment]
Merged in #2843
